### PR TITLE
Add beta factor to evaporation

### PR DESCRIPTION
**Pull request by szy21**
## Purpose 

Co-authoured-by: @LenkaNovak @kmdeck 
Adds a beta factor to evaporation when surface condition is `ValuesOnly` or `Coefficients` to account for surface resistance. 

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/SurfaceFluxes.jl [hunk 1]
```diff
@@ -196,6 +196,7 @@ Base.@kwdef struct Coefficients{FT, VI, VS} <: AbstractSurfaceConditions{FT, VI,
     z0m::FT
     z0b::FT
     gustiness::FT = FT(1)
+    beta::FT = FT(1)
```
Comment (akshaysridhar) [inline src/SurfaceFluxes.jl:199]: Is it necessary to also add this to the `Fluxes` / `FluxesandFrictionVelocity` methods? In the current call to `surface_conditions`, it seems, if `Fluxes` or `FluxesandFrictionVelocity` are known, then:
(1) SHF, LHF are known, [SHF uses the evaporation function, so known SHF implies that the \beta factor, or land surface resistance is known]
(2) Ch is computed from the Monin-Obukhov length 
(3) ` -ρ_sfc * Ch * windspeed(sc) * Δqt(param_set, sc) * sc.beta` still gives the evaporation right ? With some known `beta` dependent on the surface? i.e. here : https://github.com/CliMA/SurfaceFluxes.jl/blob/60fd6e0c01d15c76e2c83c8ce1d73dcf763bcf1d/src/SurfaceFluxes.jl#L306?

Comment (akshaysridhar) [reply src/SurfaceFluxes.jl:199]: If `Fluxes` are known then we have https://github.com/CliMA/SurfaceFluxes.jl/blob/60fd6e0c01d15c76e2c83c8ce1d73dcf763bcf1d/src/SurfaceFluxes.jl#L722, but it seems we might still need the `beta` factor for line 306 - is this correct ?

File: src/SurfaceFluxes.jl [hunk 2]
```diff
@@ -759,6 +754,39 @@ function latent_heat_flux(param_set, Ch::FT, sc::Union{ValuesOnly, Coefficients}
...
+    thermo_params = SFP.thermodynamics_params(param_set)
+    ρ_sfc = TD.air_density(thermo_params, ts_sfc(sc))
+    return -ρ_sfc * Ch * windspeed(sc) * Δqt(param_set, sc)
+end
+
+"""
+    evaporation(param_set, Ch, sc)
+
+Compute and return the evaporation. When `sc` is `ValuesOnly` or `Coefficients`, a `beta` factor
+is used to represent the resistance of the surface.
```
Comment (akshaysridhar) [inline src/SurfaceFluxes.jl:777]: Thanks for updating the docstrings as well.

File: src/SurfaceFluxes.jl (hunk 1 above)
Comment (akshaysridhar) [reply src/SurfaceFluxes.jl:199]: Resolved in latest commit ( added dispatch for `evaporation` function given latent heat fluxes)

File: src/SurfaceFluxes.jl [hunk 3]
```diff
@@ -759,6 +754,46 @@ function latent_heat_flux(param_set, Ch::FT, sc::Union{ValuesOnly, Coefficients}
...
+evaporation is directly calculated from the latent heat flux.
+## Arguments
+  - param_set: Abstract Parameter Set containing physical, thermodynamic parameters.
+  - Ch: Thermal exchange coefficient
+  - sc: Container for surface conditions based on known combination
+        of the state vector, and {fluxes, friction velocity, exchange coefficients} for a given experiment
+"""
+function evaporation(param_set, sc::Union{Fluxes, FluxesAndFrictionVelocity}, Ch)
+    thermo_params = SFP.thermodynamics_params(param_set)
+    grav::FT = SFP.grav(param_set)
```
Comment (akshaysridhar) [inline src/SurfaceFluxes.jl:770]: FT - type is undefined here I think

Comment (szy21) [reply src/SurfaceFluxes.jl:770]: oops, thanks!

File: src/SurfaceFluxes.jl [hunk 4]
```diff
@@ -759,6 +754,46 @@ function latent_heat_flux(param_set, Ch::FT, sc::Union{ValuesOnly, Coefficients}
...
+
+Compute and return the evaporation. When `sc` is `Fluxes` or `FluxesAndFrictionVelocity`,
+evaporation is directly calculated from the latent heat flux.
+## Arguments
+  - param_set: Abstract Parameter Set containing physical, thermodynamic parameters.
+  - Ch: Thermal exchange coefficient
+  - sc: Container for surface conditions based on known combination
+        of the state vector, and {fluxes, friction velocity, exchange coefficients} for a given experiment
+"""
+function evaporation(param_set, sc::Union{Fluxes, FluxesAndFrictionVelocity}, Ch)
```
Comment (akshaysridhar) [inline src/SurfaceFluxes.jl:768]:
```suggestion
function evaporation(param_set, sc::Union{Fluxes, FluxesAndFrictionVelocity}, Ch::FT) where {FT}
```

**Review by akshaysridhar — APPROVED** [review]
LGTM! Please rebase before merging. Thanks!

**Pull request merged**